contexts/AutocratContext.tsx

```diff
@@ -76,18 +76,39 @@ export function AutocratProvider({ children }: { children: ReactNode }) {
 
   const { program: openbookTwap } = useOpenbookTwap();
   const [daoState, setDaoState] = useState<DaoState>();
-  const [proposals, setProposals] = useState<ProposalAccountWithKey[]>();
-  console.log(programVersion, proposals);
+  const [proposals, setProposals] = useState<Proposal[]>();
 
   const fetchState = useCallback(async () => {
     setDaoState(await autocratProgram.account.dao.fetch(dao));
   }, [autocratProgram, dao]);
 
   const fetchProposals = useCallback(async () => {
-    const props = ((await autocratProgram?.account.proposal.all()) || []).sort((a, b) =>
+    const props = ((await autocratProgram?.account.proposal?.all()) || []).sort((a, b) =>
       a.account.number < b.account.number ? 1 : -1,
     );
-    setProposals(props);
+
+    let _proposals: Proposal[] = props.map((prop) => ({
```

Comment:
why not directly put the result of this in the `setProposals` ?

Reply:
The proposals texts take a bit of time to fetch, I wanted to not sacrifice the speed